src/main/java/duke/command/DeletePatientTaskCommand.java

```diff
@@ -66,31 +67,33 @@ public void execute(PatientTaskList patientTaskList, TaskManager tasks, PatientM
         if (patientId != 0) {
             try {
                 Patient toBeDeletedPatient = patientManager.getPatient(patientId);
-                for (PatientTask patientTask: patientTaskList.getPatientTask(patientId)) {
-                    if (patientTask.getUid() == taskId) {
-                        patientTaskList.deletePatientTaskByUniqueId(taskId);
-                        patientTaskStorage.save(patientTaskList.fullPatientTaskList());
-                        ui.patientTaskDeleted(patientTask, toBeDeletedPatient);
-                    }
+                if (patientTaskList.isPatientIdExist(patientId)) {
+                    patientTaskList.deleteEntirePatientTask(patientId);
+                    patientTaskStorage.save(patientTaskList.fullPatientTaskList());
+                    ui.patientTaskAllDeleted(patientManager.getPatient(patientId));
                 }
             } catch (DukeException e) {
-                throw new DukeException("Task or patient is not found!" + e.getMessage());
+                throw new DukeException("Patient is not found!");
+            }
+        } else if (taskId != 0) {
+            try {
+                patientTaskList.deletePatientTaskByUniqueId(taskId);
+                patientTaskStorage.save(patientTaskList.fullPatientTaskList());
+                ui.patientTaskDeleted(taskId);
+            } catch (DukeException e) {
+                throw new DukeException("Task is not found!");
             }
         } else {
             try {
                 String patientName = this.deletedPatientInfo;
                 ArrayList<Patient> patientsWithSameName = patientManager.getPatientByName(patientName);
-                ArrayList<PatientTask> toBeDeleted =
-                        patientTaskList.getPatientTask(patientsWithSameName.get(0).getID());
-                for (PatientTask patientTask: toBeDeleted) {
-                    if (patientTask.getUid() == taskId) {
-                        patientTaskList.deletePatientTaskByUniqueId(taskId);
-                        patientTaskStorage.save(patientTaskList.fullPatientTaskList());
-                        ui.patientTaskDeleted(patientTask, patientsWithSameName.get(0));
-                    }
+                if (patientTaskList.isPatientIdExist(patientsWithSameName.get(0).getID())) {
```

Comment:
`.get(0)` should be renamed for the sake of clarity. It's hard to tell what it's supposed to do based on the name. 

Also, using a 'magic number' such as 0 can cause problems with understanding the value's purpose in the future. Might be better to assign it to a variable with a meaningful name.

Reply:
solved